File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/responses/notification/NotificationResponse.java
package com.example.dalathasfarm.responses.notification;

import com.example.dalathasfarm.models.Notification;
import com.example.dalathasfarm.responses.BaseResponse;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class NotificationResponse extends BaseResponse {
    @JsonProperty("notification_id")
    private Integer id;

    @JsonProperty("user_id")
    private Integer userId;

    private String title;
    private String content;
    private String type;

    @JsonProperty("is_read")
    private Boolean isRead;

    public static NotificationResponse fromNotification (Notification notification) {
        NotificationResponse newNotificationResponse = NotificationResponse.builder()
                .id(notification.getId())
                .userId(notification.getUser().getId())
                .title(notification.getTitle())
                .content(notification.getContent())
                .type(notification.getType().name())
                .isRead(notification.getIsRead())
                .build();
        newNotificationResponse.setCreatedAt(notification.getCreatedAt());
        newNotificationResponse.setUpdatedAt(notification.getUpdatedAt());

        return newNotificationResponse;
    }
}

File: android/lib/responses/notification_response.dart
class NotificationResponse {
  final int notification_id;
  final int user_id;
  final String title;
  final String content;
  final String type;
  final bool is_read;
  final DateTime createdAt;
  final DateTime updatedAt;

  NotificationResponse({
    required this.notification_id,
    required this.user_id,
    required this.title,
    required this.content,
    required this.type,
    required this.is_read,
    required this.createdAt,
    required this.updatedAt,
  });

  factory NotificationResponse.fromJson(Map<String, dynamic> json) {
    return NotificationResponse(
      notification_id: json['notification_id'],
      user_id: json['user_id'],
      title: json['title'],
      content: json['content'],
      type: json['type'],
      is_read: json['is_read'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}
